src/query/procedure/module.cpp

```diff
@@ -638,6 +645,16 @@ void ModuleRegistry::UnloadAllModules() {
 
 utils::MemoryResource &ModuleRegistry::GetSharedMemoryResource() noexcept { return *shared_; }
 
+bool ModuleRegistry::RegisterMgProcedure(const std::string_view name, mgp_proc proc) {
+  std::unique_lock<utils::RWLock> guard(lock_);
+  if (auto module = modules_.find("mg"); module != modules_.end()) {
+    auto *builtin_module = dynamic_cast<BuiltinModule *>(module->second.get());
+    builtin_module->AddProcedure(name, std::move(proc));
+    return true;
+  }
+  return false;
```

Comment:
```suggestion
  auto module = modules_.find("mg");
  if (module == modules_.end()) {
    return false;
  }
  auto *builtin_module = dynamic_cast<BuiltinModule *>(module->second.get());
  builtin_module->AddProcedure(name, std::move(proc));
  return true;
```
Maybe like this? :smile:

Reply:
I am not sure about this. I personally like that the variable is local to the scope on my version but I am happy to make  the change if you find this snippet more readable.

Reply:
It's fine, wanted to hear your opinion.